pkg/kubectl/cmd/util/factory.go

```diff
@@ -274,12 +274,62 @@ func makeInterfacesFor(versionList []unversioned.GroupVersion) func(version unve
 	}
 }
 
+type DiscoveryExpander struct {
```

Comment:
Ack @deads2k i just started rebase etc. will have it ready hopefully later today